sys/include/ztimer64.h

```diff
@@ -0,0 +1,519 @@
+ * The function will return after waiting at most @p timeout ticks.
+ *
+ * @note: This might function might leave a message with type MSG_ZTIMER64 in the
+ *        thread's message queue, which must be handled (ignored).
+ *
+ * @param[in]   clock           ztimer clock to operate on
+ * @param[out]  msg             pointer to buffer which will be filled if a
+ *                              message is received
+ * @param[in]   timeout         relative timeout, in @p clock time units
+ *
+ * @return  >=0 if a message was received
+ * @return  -ETIME on timeout
+ */
+static inline int ztimer64_msg_receive_timeout(ztimer64_clock_t *clock,
+                                               msg_t *msg,
+                                               uint64_t timeout)
+{
+    return ztimer64_msg_receive_until(clock, msg,
+                                      ztimer64_offset2absolute(clock, timeout));
+}
+
+#define MSG_ZTIMER64 0xc83f   /**< msg type used by ztimer64_msg_receive_timeout */
+
+/**
+ * @brief Suspend the calling thread until the time (@p last_wakeup + @p period)
+ *
+ * This function can be used to create periodic wakeups.
+ *
+ * When the function returns, @p last_wakeup is set to
+ * (@p last_wakeup + @p period).
+ *
+ * @c last_wakeup should be set to ztimer64_now(@p clock) before first call of the
+ * function.
+ *
+ * If the time (@p last_wakeup + @p period) has already passed, the function
+ * sets @p last_wakeup to @p last_wakeup + @p period and returns immediately.
+ *
+ * @note if you period is smaller than 2^32 ticks, consider using the 32bit version.
+ *
+ * @param[in]   clock           ztimer clock to operate on
+ * @param[in]   last_wakeup     base time stamp for the wakeup
+ * @param[in]   period          time in ticks that will be added to @p last_wakeup
+ */
+void ztimer64_periodic_wakeup(ztimer64_clock_t *clock, uint64_t *last_wakeup,
+                              uint64_t period);
+
+/**
+ * @brief   Put the calling thread to sleep until the specified time
+ *
+ * @param[in]   clock           ztimer clock to use
+ * @param[in]   target          wakeup time, in @p clock time
+ */
+void ztimer64_sleep_until(ztimer64_clock_t *clock, uint64_t target);
+
+/**
+ * @brief   Put the calling thread to sleep for the specified number of ticks
+ *
+ * @param[in]   clock           ztimer clock to use
+ * @param[in]   duration        duration of sleep, in @p ztimer time units
+ */
+static inline void ztimer64_sleep(ztimer64_clock_t *clock, uint64_t duration)
+{
+    ztimer64_sleep_until(clock, ztimer64_offset2absolute(clock, duration));
+}
+
+/**
+ * @brief   Busy-wait until specified target time
+ *
+ * @note: This blocks lower priority threads. Use only for *very* short delays.
+ *
+ * @param[in]   clock           ztimer clock to use
+ * @param[in]   target          time when spinning should end, in @p clock time
+ */
+static inline void ztimer64_spin_until(ztimer64_clock_t *clock, uint64_t target)
```

Comment:
Should this function be removed? Or is there a use case for 64 bit spin times?

Reply:
well, spinning until absolute time has a use, as it's difficult to overspin, and the arithmetic is simple.
and, there's "All ztimer_* functions have a ztimer64_* equivalent that behaves the same, but allowing offsets >= 2^32".

Reply:
But is there really a use case for spinning longer than 2^32 microseconds, or more than an hour. 

IMHO this doesn't serve a purpose and I would leave it out and add a remark to the quote you mention. The whole idea of ztimer64 conflicts with the "Use only for *very* short delays" mentioned in the doxygen of this function.

Reply:
no, the point is that ztimer32 doesn't allow absolute times, so as soon as a target time is coming close, using relative timeouts becomes fiddly.
With absolute times, you can calculate your target time once, do stuff, then spin with the same time, and be sure it never underflows. (depending on the hardware, 64bit calculations, conversion, hw timer read make this not a good choice, but hey...).

Reply:
Like, if we actually do support 64bit backends for ztimer64 at some point, like the ptp stuff on stm32, there'd be a native 64bit timer chain.
IMO that can be useful.

Reply:
Ok, convinced me enough. Thanks for removing the regular version.